### PR TITLE
Federate edge image list

### PR DESCRIPTION
**Pull request by acosferreira**
Initial version for the parity between image builder and edge image list

### CODE REVIEW

**Comment by acosferreira** [comment]
FYI @mgold1234 @matoval @loadtheaccumulator This is the initial setup for the parity.
@croissanne The feature flag and layout approach is under the other pr's opened by Michal, please let us know if you need any changes on our side

File: src/Components/ImagesTable/ImagesTable.js [hunk 1]
```diff
@@ -1,5 +1,8 @@
 import React, { useEffect, useState } from 'react';
 
+import AsyncComponent from '@redhat-cloud-services/frontend-components/AsyncComponent';
+
```
Comment (croissanne) [inline src/Components/ImagesTable/ImagesTable.js]: no newline here

File: src/Components/ImagesTable/ImagesTable.js [hunk 2]
```diff
@@ -26,7 +29,9 @@ import {
 } from '@patternfly/react-table';
 import PropTypes from 'prop-types';
 import { useDispatch, useSelector } from 'react-redux';
-import { Link, useNavigate } from 'react-router-dom';
+
```
Comment (croissanne) [inline src/Components/ImagesTable/ImagesTable.js]: no newline here, should be 1 import block

**Review by croissanne — CHANGES_REQUESTED** [review]
overall lgtm but can you clean up the git history a bit?

1. no merge commits, we strictly rebase
2. please run the linter on each commit individually, this avoids noise in later commits
3. read https://www.osbuild.org/guides/developer-guide/workflow.html to get more information about the format of commits and PRs

File: src/Components/ImagesTable/ImagesTable.js [hunk 3]
```diff
@@ -154,6 +158,37 @@ const ImagesTable = () => {
   const itemsStartInclusive = (page - 1) * perPage;
   const itemsEndExclusive = itemsStartInclusive + perPage;
 
+  const notificationProp = {
```
Comment (croissanne) [inline src/Components/ImagesTable/ImagesTable.js:161]: https://github.com/RedHatInsights/image-builder-frontend/pull/1160 seems to add a copy of this, maybe just put this prop in `src/Utilities/edge.js` directly ? When 1160 gets rebased we won't have duplicate code then.

File: src/Components/ImagesTable/ImagesTable.js [hunk 4]
```diff
@@ -24,10 +25,11 @@ import {
   Thead,
   Tr,
 } from '@patternfly/react-table';
+import AsyncComponent from '@redhat-cloud-services/frontend-components/AsyncComponent';
+import { addNotification } from '@redhat-cloud-services/frontend-components-notifications/redux';
 import PropTypes from 'prop-types';
 import { useDispatch, useSelector } from 'react-redux';
-import { Link, useNavigate } from 'react-router-dom';
-
+import { Link, useNavigate, useLocation } from 'react-router-dom';
```
Comment (croissanne) [inline src/Components/ImagesTable/ImagesTable.js:32]: Ah sorry, I was wrong, the linter complains. There's 1 newline needed between the external and internal imports.

**Review by croissanne — CHANGES_REQUESTED** [review]
Thanks! 2 small comments otherwise lgtm.

File: src/Components/ImagesTable/ImagesTable.js (hunk 3 above)
Comment (acosferreira) [reply src/Components/ImagesTable/ImagesTable.js:161]: @croissanne the src/Utilities/edge.js was added on #1160 can we do the clean up when rebase?

**Review by mgold1234 — COMMENTED** [review]
@croissanne those changes appear in my pr, maybe let's merge my pr first with the correct tab and location of the button and then continue to add there the implementation of the federation module that Ana did?

**Comment by acosferreira** [comment]
@croissanne I've addressed some of your comments, and as @mgold1234 mentioned, the other layout changes are on ther pr.
Let us know the better way to proceed

**Comment by acosferreira** [comment]
image

**Comment by acosferreira** [comment]
image

**Comment by croissanne** [comment]
> image
> 

Hmm, that still doesn't have the white background for the tab headers.

Also I think every PR should make sense and work  as a unit. So we can either rebase the other on top of this. Or we can merge them one by one, but both need to be green of course.

**Comment by acosferreira** [comment]
@mgold1234 lets rebase our prs and get in one then we can have the complete flow working.
@croissanne the tests that is broken is related to the tab changes we are looking to fix that based on the feature flag.

**Comment by acosferreira** [comment]
Closing this pr, once the changes were rebased on #1149

**Pull request closed without merging**